v2/audit/conn/hang_conn.go

```diff
@@ -0,0 +1,46 @@
+package conn
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/microsoft/go-otel-audit/v2/audit/msgs"
+)
+
+// TestHangConn is a test connection that hangs on CloseSend to demonstrate a hanging bug.
+type TestHangConn struct {
```

Comment:
Not seeing where this is used in this PR which I assume means it's intentional that this is exported (useful for callers to test with maybe?), but just checking